### ISSUE TITLE
Fix Bruker plane specification implementation through file_paths

### ISSUE THREAD

**Issue by h-mayorquin** [ben-dichter-consulting/traceextractors#345]
[Draft]

I discussed this today with @weiglszonja and we converged to this solution. I am writing the context here for provenance and as a to-do list. 

The main problem is a clash between the plane specification and the channel specification that are both passed as `stream_name` to `BrukerTiffSinglePlaneImagingExtractor`.  **For the test examples we have got this works because the channel names are always part of the file name ** (and therefore of the string implementation) see here: 

https://github.com/catalystneuro/roiextractors/blob/29c3491913f4be558a950262fd854a445a6b2621/src/roiextractors/extractors/tiffimagingextractors/brukertiffimagingextractor.py#L167-L175

And this is applied to xml elements that look like this:

```xml
      <File channel="2" channelName="Ch2" filename="NCCR32_2022_11_03_IntoTheVoid_t_series-005_Cycle00001_Ch2_000001.ome.tif" />
```

See that the `channelName` is part of the file_paths. However, this assumption does not hold for data from the Clandinin lab and the data from this other issue https://github.com/catalystneuro/roiextractors/issues/341 here so we need to decouple.

A first attempt was done here:
https://github.com/catalystneuro/roiextractors/pull/343

This works for the `BrukerTiffSinglePlaneImagingExtractor` because we can look for the file_paths that correspond to a channel name and have very specific list of file and that works. Unfortunately, this breaks the use case of  `BrukerTiffMultiPlaneImagingExtractor` passing plane streams to `BrukerTiffSinglePlaneImagingExtractor` here:

https://github.com/catalystneuro/roiextractors/blob/29c3491913f4be558a950262fd854a445a6b2621/src/roiextractors/extractors/tiffimagingextractors/brukertiffimagingextractor.py#L239-L242 

Because that also relied **on the assumption that the plane specification we have is contained in the file name.**

So what gives? 

The thing that we can do programatically is find all the file_paths that correspond to a channel.
The thing that we can do programatically is find a simple and friendly user specifation to get a single plane. Yet.

**Comment by weiglszonja** [comment]
Yeah we need the number of planes as well, if we have the list of file_paths maybe we can do something similar as in:
https://github.com/catalystneuro/roiextractors/blob/29c3491913f4be558a950262fd854a445a6b2621/src/roiextractors/extractors/tiffimagingextractors/brukertiffimagingextractor.py#L158
I think in that example we have two planes: ..._Ch2_000002.ome.tif and ..._Ch2_000001.ome.tif 
maybe from the list of file paths we could find the part until the last "_" to determine how many unique files we have and that should be the number of planes? 
But again this assumption would only hold true when we have volumetric data, as in [this example](https://gin.g-node.org/CatalystNeuro/ophys_testing_data/src/main/imaging_datasets/BrukerTif/NCCR32_2022_11_03_IntoTheVoid_t_series-005) there is a simple plane.

**Comment by h-mayorquin** [comment]
Yes, to get the number of planes, I think we can count Sequence elements in the xml. What do you think?

For choosing a single plane though we still need to use some sort of regular expression as you were doing.

**Comment by h-mayorquin** [comment]
PR #567 sidesteps this and should be the way to move forward. Closing this.
